### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by the-real-not-a-number**
added momentary action to 'on/off switch'

### CODE REVIEW

File: gpio-adapter.js [hunk 1]
```diff
@@ -66,6 +66,23 @@ class GpioProperty extends Property {
           this.device.notifyPropertyChanged(this);
         }
       });
+      if (this.device.pinConfig.unlatch) {
+        setTimeout(() => {
+          value = 0;
```
Comment (mrstegeman) [inline gpio-adapter.js]: You're setting `value = 0` here, which means the ternary in the next line is meaningless.

File: manifest.json [hunk 2]
```diff
@@ -20,8 +20,11 @@
         {
           "activeLow": false,
           "direction": "out",
+          "latching": true,
```
Comment (mrstegeman) [inline manifest.json]: This seems to be unused.

File: package.json [hunk 3]
```diff
@@ -105,6 +107,13 @@
               },
               "activeLow": {
                 "type": "boolean"
+              },
+              "unlatch": {
+                "type": "boolean"
+              },
+              "unlatchDelay": {
+                "type": "number",
+                "minimum": 0
```
Comment (mrstegeman) [inline package.json:116]: This minimum is missing from the `package.json` patch above.

File: package.json [hunk 4]
```diff
@@ -105,6 +107,13 @@
               },
               "activeLow": {
                 "type": "boolean"
+              },
+              "unlatch": {
```
Comment (mrstegeman) [inline package.json]: I know the others don't already have this, but can you add a meaningful `title` and `description` to the two new config items, both here and in `manifest.json`?

**Review by mrstegeman — CHANGES_REQUESTED** [review]
Please also restore the deleted files.

**Comment by the-real-not-a-number** [comment]
requested amendments effected

**Comment by mrstegeman** [comment]
There are no new commits...

File: gpio-adapter.js.js [hunk 5]
```diff
@@ -66,6 +66,22 @@ class GpioProperty extends Property {
           this.device.notifyPropertyChanged(this);
         }
       });
+      if (this.device.pinConfig.momentaryButton) {
+        setTimeout(() => {
+          this.device.gpio.write(value = 0, (err) => {
+            if (err) {
+              console.error('GPIO: write for pin:', this.device.name, 'failed');
+              console.error(err);
+              reject(err);
```
Comment (mrstegeman) [inline gpio-adapter.js.js]: You're attempting to reject/resolve the same promise twice here. You can just omit the `reject(err)` here and the `resolve(this.value)` below.

File: package.json [hunk 6]
```diff
@@ -53,7 +53,14 @@
           "name": "led",
           "direction": "out",
           "value": 0,
-          "activeLow": false
+          "activeLow": false,
+<<<<<<< HEAD
```
Comment (mrstegeman) [inline package.json]: Please resolve these merge errors.

File: package.json [hunk 7]
```diff
@@ -18,7 +18,7 @@
   "license": "MPL-2.0",
   "repository": {
     "type": "git",
-    "url": "https://github.com/mozilla-iot/gpio-adapter.git"
+    "url": "git+https://github.com/mozilla-iot/gpio-adapter.git"
```
Comment (mrstegeman) [inline package.json:21]: This change is unnecessary

**Review by mrstegeman — CHANGES_REQUESTED** [review]

**Comment by mrstegeman** [comment]
Looks like you've now deleted `package.json` all together and changed the permissions on `package.sh`.

File: gpio-adapter.js [hunk 8]
```diff
@@ -66,6 +66,22 @@ class GpioProperty extends Property {
           this.device.notifyPropertyChanged(this);
         }
       });
+      if (this.device.pinConfig.momentaryButton) {
+        setTimeout(() => {
+          this.device.gpio.write(value = 0, (err) => {
+            if (err) {
+              console.error('GPIO: write for pin:', this.device.name, 'failed');
+              console.error(err);
+              reject(err);
```
Comment (mrstegeman) [inline gpio-adapter.js:75]: Remove the reject and resolve below, they’re unnecessary

**Review by mrstegeman — CHANGES_REQUESTED** [review]
Please also revert the file permissions on `package.sh`.

**Pull request closed without merging**